donations/views.py

```diff
@@ -0,0 +1,94 @@
+import json
+import base64
+import requests
+import urlparse
+from django.shortcuts import render
+from django.core.urlresolvers import reverse
+from django.conf import settings
+from django.http import HttpResponse, JsonResponse
+from django.contrib.sites.models import Site
+from django.contrib.auth.models import User
+from django.views.generic import ListView
+from django.views.decorators.csrf import csrf_exempt
+from models import Donation, DonationCampaign
+
+
+@csrf_exempt
+def donation_complete(request):
+    '''This view listens to a notification made from paypal when someone makes
+    a donation, it validates the data and then stores the donation.
+    '''
+    params = {'cmd': '_notify-validate'}
+    for key, value in request.POST.items():
+        params[key] =  value
+    if settings.DEBUG:
+        paypal_validation_url = "https://www.sandbox.paypal.com/cgi-bin/webscr"
+    else:
+        paypal_validation_url = "https://www.paypal.com/cgi-bin/webscr"
+    req = requests.post(paypal_validation_url, data=params)
+    if req.text == 'VERIFIED':
+        extra_data = json.loads(base64.b64decode(params['custom']))
+        campaign = DonationCampaign.objects.get(id=extra_data['campaign_id'])
+        user = None
+        if extra_data['user_id']:
```

Comment:
If the value of user_id is None then is an anonymous donation, do you think it would be better to not send the user_id key in that case?